base/show.jl

```diff
@@ -1419,7 +1469,8 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
         end
         print(io, '"')
 
-    elseif (head === :&#= || head === :$=#) && nargs == 1
+    # elseif (head === :&#= || head === :$=#) && nargs == 1
```

Comment:
Me too I'm a little bit confused about this issue. In any case, I've added two test case for nested quotes and interpolations, which pass, but I'm not sure that this matter is completely addressed.

Reply:
I read through that again. Vaguely, I think the problem is that the fallback for printing `Expr`s with unknown head (or other non-surface syntax) is currently to interpolate in an unquoted `Expr` constructor with the arguments to that re-quoted again. Then we run into trouble for `$`s in the re-quoted section. I haven't got an example yet though, and it looks like the expander (`julia-bq-expand` if I'm not mistaken) has changed since then.

Reply:
I've added some test cases to check for possible misbehavior in this scenario, and things seam
to be ok. In the process I've found a problem with the printing of `$x` and the like. More on that issue and its solution in another comment.